OutOfSchool/OutOfSchool.WebApi/Services/WorkshopServicesCombiner.cs

```diff
@@ -119,20 +119,23 @@ public async Task<SearchResult<WorkshopCard>> GetByFilter(WorkshopFilter filter)
         }
 
         /// <inheritdoc/>
-        public async Task<IEnumerable<WorkshopDTO>> GetByProviderId(long id)
+        public async Task<IEnumerable<WorkshopCard>> GetByProviderId(long id)
         {
-            var workshop = await databaseService.GetByProviderId(id).ConfigureAwait(false);
+            var workshops = await databaseService.GetByProviderId(id).ConfigureAwait(false);
 
-            return workshop;
+            var workshopCards = DtoModelsToWorkshopCards(workshops);
+
+            return workshopCards;
         }
 
         private List<WorkshopCard> DtoModelsToWorkshopCards(IEnumerable<WorkshopDTO> source)
         {
             List<WorkshopCard> workshopCards = new List<WorkshopCard>();
-            foreach (var item in source)
+
+            Parallel.ForEach(source, currentElement =>
```

Comment:
`Parallel.ForEach` seems to be bit excessive here

Reply:
You could use instead simple 
`var workshopCards = source.Select(currentElement => currentElement.ToCardDto()).ToList()`

Reply:
1. Writing to List{T} [is not thread safe](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1?view=net-5.0#thread-safety)
2. Be aware that parallelization here will change items order inside the collection so it will be hard to add a pagination functionality in the future.
If you still want to use parallelization here please use thread safe collection or try something like 
`=> source.AsParallel()` ... `code from @VadymLevkovskyi  comment`

Reply:
@h4wk13 please be aware, that I'm not suggesting to use `ToList()` inside `Parallel.ForEach()`. I was talking about total replacing of `Parallel.ForEach()` by simple `Select()`.

Reply:
Discussion seems to be outdated, and nothing is touched, so I set this thread to resolved